### DISCUSSION TITLE
LDT ensemble restart generation

### DISCUSSION THREAD

**Discussion by nchawang** [NASA-LIS/LISF#979] [Q&A] [answered]
Hi,
I was trying to generate an Ensemble Restart file, i had the lis_input file and the open loop restart file. while running LDT, I get an error as the following:
**"Fatal error in PMPI_Allgather: Invalid buffer pointer, error stack:
PMPI_Allgather(961): MPI_Allgather(sbuf=0xbc882a0, scount =1, MPI_INTEGER, rbuf=0xbc882a0,rcount=1,MPI_INTEGER, MPI_COMM_WORLD) failed
PMPI_Allgather(936): Buffers must not be aliased"**

I have attached the lis_input.nc file, OL restart file, ldt.config and ldtlog.0000 file. Have I missed something? I was following testcase 3 available at: https://lis.gsfc.nasa.gov/lis-testcases

Thanks
[error_ensemgen.zip](https://github.com/NASA-LIS/LISF/files/8023070/error_ensemgen.zip)

**Comment by dmocko** [comment, marked as answer]
Hi @nchawang,

Are you still encounting this error?

If so, can you please go into LISF/ldt/make, type "gmake realclean", and then re-compile/re-run LDT?

This time when you configure, select "0" for this option:

`Parallelism (0-serial, 1-dmpar, default=0):  0`

It's possible that running LDT parallel for this step of the testcase will give you this error.

David

**Reply by nchawang** [reply]
Hi @dmocko 
Compiling it in serial fixed the error. 
This issue only arises in ensemble generation though, normal Open loop LDT run does not have this error crop up in parallel.
Thanks!
Noel C.

**Reply by Hyunglok-Kim** [reply]
Hi David,

@dmocko,
I'm experiencing the exact same issue as Noel. I encountered this problem when I attempted testcase step 3 (enrst) with LISF 7.4.2.

Thanks,
Hyung
